extension/persistence/eclipselink/src/main/java/org/apache/polaris/extension/persistence/impl/eclipselink/DiscoverableMetaStoreManagerFactory.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.extension.persistence.impl.eclipselink;
+
+import io.dropwizard.jackson.Discoverable;
+import io.dropwizard.jackson.DiscoverableSubtypeResolver;
+
+/**
+ * Discoverability aid for Dropwizard. Leaf classes cannot implement {@link Discoverable} directly,
+ * they have to implement another interface, which implements {@link Discoverable}.
+ *
+ * @see DiscoverableSubtypeResolver
+ */
+public interface DiscoverableMetaStoreManagerFactory extends Discoverable {}
```

Comment:
This doesn't make a lot of sense to me. When the subtype resolver is asked for implementations, it'll be for a common interface - i.e., `MetaStoreManagerFactory` in this case. Why would the subtype resolver ever ask for the eclipse-link specific version of the `DiscoverableMetaStoreManagerFactory`?

Reply:
Type resolution during YAML parsing is handled by Jackson, and it properly recognizes all types in the java hierarchy (even those not implementing `Discoverable`).

However, Jackson (in this setup) relies on `DiscoverableSubtypeResolver` to discover implementation classes. That latter part has peculiar logic, where `DiscoverableSubtypeResolver` performs a two stage lookup: 1) classes listed in `META-INF/services/io.dropwizard.jackson.Discoverable`, 2) classes listed in the service manifests for the classes from step 1. I do not really know why it does not directly inject classes from step 1 into Jackson.

I agree, it looks odd. However, having one `Discoverable` interface is not possible as this PR specifically proposes to remove DW dependencies from `polaris-core`, where the base interface exists and there is not other common root between the "in memory" metastore implementation and the EclipseLink implementation. We could add a rather small common module for that, of course, but in my mind having two separate `Discoverable` interfaces was less intrusive. WDYT?

Reply:
> Leaf classes cannot implement {@link Discoverable} directly, they have to implement another interface, which implements {@link Discoverable}.

Isn't this the exact setup we have right now, where the leaf classes implement `MetastoreManagerFactory`?

Should we just move `MetastoreManagerFactory` out of core?

Reply:
> Should we just move MetastoreManagerFactory out of core?

That's an option. This is what I meant by "add a rather small common module" above, because the EclipseLink and In-Memory implementations have to share it.

Would do other people think? @collado-mike @adutra ?

Reply:
So, if I get this correctly, we would all love to have `MetastoreManagerFactory` directly in polaris-service like many other similar beans, but we can't, because of EclipseLink.

If that is correct, I'd prefer to keep your current solution _for now_, then revisit this later, when/if EclipseLink is replaced with some more robust persistence solution.